src/gentropy/susie_finemapper.py

```diff
@@ -0,0 +1,140 @@
+"""Step to run a finemapping using."""
+
+from __future__ import annotations
+
+from typing import Any
+
+import numpy as np
+import pyspark.sql.functions as f
+from pyspark.sql import DataFrame, Window
+
+from gentropy.common.session import Session
+from gentropy.dataset.study_locus import StudyLocus
+
+
+class SusieFineMapperStep:
+    """SuSie finemaping. It has generic methods to run SuSie fine mapping for a study locus.
+
+    This class/step is the temporary solution of the fine-mapping warpper for the development purposes.
+    In the future this step will be refactored and moved to the methods module.
+    """
+
+    @staticmethod
+    def susie_inf_to_studylocus(
+        susie_output: dict[str, Any],
+        session: Session,
+        _studyId: str,
+        _region: str,
+        _join: DataFrame,
+        cs_lbf_thr: float = 2,
+    ) -> StudyLocus:
+        """Convert SuSiE-inf output to studyLocus DataFrame.
+
+        Args:
+            susie_output (dict[str, Any]): SuSiE-inf output dictionary
+            session (Session): Spark session
+            _studyId (str): study ID
+            _region (str): region
+            _join (DataFrame): DataFrame with variant information
+            cs_lbf_thr (float): credible set logBF threshold, default is 2
+
+        Returns:
+            StudyLocus: StudyLocus object with fine-mapped credible sets
+        """
+        variants = np.array(
+            [row["variantId"] for row in _join.select("variantId").collect()]
+        ).reshape(-1, 1)
+        PIPs = susie_output["PIP"]
+        lbfs = susie_output["lbf_variable"]
+        mu = susie_output["mu"]
+        susie_result = np.hstack((variants, PIPs, lbfs, mu))
+
+        L_snps = PIPs.shape[1]
+
+        # Extracting credible sets
+        order_creds = list(enumerate(susie_output["lbf"]))
+        order_creds.sort(key=lambda x: x[1], reverse=True)
+        cred_sets = None
+        counter = 0
+        for i, value in order_creds:
```

Comment:
`i` is fine, this is a scanonical way of calling an index variable, however `value` is not very telling. At row 110 it is very complicated what does this `value`refer to.

Reply:
renamed it for cs_lbf_value